src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -55,6 +73,100 @@ void GraphingCalculator::ViewModel::set(GraphingCalculatorViewModel^ vm)
+// data to be shared.  We will request the current graph image from teh grapher as a stream that will pass to the share request.
+void GraphingCalculator::OnDataRequested(DataTransferManager^ sender, DataRequestedEventArgs^ args)
+{
+    try
+    {
+        // Get our title from the localized resources
+        auto resourceLoader = Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView();
+        auto EmptyEquationString = resourceLoader->GetString(L"EmptyEquationString");
+
+        auto htmlResources = ref new Map<String^, RandomAccessStreamReference^>();
+        std::wstring rawHtml = L"<p><img src='graph.png'></p>";
+
+        auto equations = ViewModel->Equations;
+        rawHtml += L"<p><table cellpadding=\"10\">";
+        rawHtml += L"<col width=\"60\">";
+        rawHtml += L"<row height=\"60\">";
+        for (unsigned i = 0; i < equations->Size; i++)
+        {
+            auto expression = equations->GetAt(i)->Expression->Data();
+            auto color = equations->GetAt(i)->LineColor;
+
+            if (equations->GetAt(i)->Expression->Length() == 0)
+            {
+                expression = EmptyEquationString->Data();
+            }
+
+            rawHtml += L"<tr>";
+
+            rawHtml += L"<td style=\"background-color:rgb(";
+            rawHtml += color.R.ToString()->Data();
+            rawHtml += L",";
+            rawHtml += color.G.ToString()->Data();
+            rawHtml += L",";
+            rawHtml += color.B.ToString()->Data();
+            rawHtml += L"); \">";
+            rawHtml += L"&#402;(&#215;)";
+            rawHtml += L"</td>";
+            rawHtml += L"<td>";
+            rawHtml += expression;
+            rawHtml += L"</td>";
+
+            rawHtml += L"</tr>";
+        }
+        rawHtml += L"</table></p>";
+
+        // Shortcut to the request data
+        auto requestData = args->Request->Data;
+
+        DataPackage^ dataPackage = ref new DataPackage();
+        auto html = HtmlFormatHelper::CreateHtmlFormat(ref new String(rawHtml.c_str()));
+
+        auto titleString = resourceLoader->GetString(L"ShareActionTitle");
+        requestData->Properties->Title = titleString;
+
+        requestData->SetHtmlFormat(html);
+
+        htmlResources->Insert(ref new String(L"graph.png"), TheGrapher->GetGraphBitmapStream());
+
+        for (auto pair : htmlResources)
+        {
+            auto k = pair->Key;
+            auto v = pair->Value;
+            requestData->ResourceMap->Insert(k, v);
+        }
+    }
+    catch(...)
+    {
+        // Something went wrong, notify the user.
+        auto resourceLoader = Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView();
+
+        auto errorTitleString = resourceLoader->GetString(L"ShareActionErrorMessage");
+        auto errorOkString = resourceLoader->GetString(L"ShareActionErrorOk");
+        auto errDialog = ref new Windows::UI::Xaml::Controls::ContentDialog();
```

Comment:
Let's sync with PM about using a Content Dialog vs another control to show the error.

Reply:
Already spoke to PM about this.  Goign with the ContentDialog for now.